packages/core/src/components/widgets/SelectWidget.js

```diff
@@ -1,49 +1,47 @@
 import React from "react";
 import PropTypes from "prop-types";
+import _isEqual from "lodash/isEqual";
 
-import { asNumber, guessType } from "../../utils";
-
-const nums = new Set(["number", "integer"]);
-
-/**
- * This is a silly limitation in the DOM where option change event values are
- * always retrieved as strings.
- */
-function processValue(schema, value) {
-  // "enum" is a reserved word, so only "type" and "items" can be destructured
-  const { type, items } = schema;
-  if (value === "") {
+function getValue(event, enumOptions, multiple) {
+  if (event.target.value === "") {
     return undefined;
-  } else if (type === "array" && items && nums.has(items.type)) {
-    return value.map(asNumber);
-  } else if (type === "boolean") {
-    return value === "true";
-  } else if (type === "number") {
-    return asNumber(value);
-  }
-
-  // If type is undefined, but an enum is present, try and infer the type from
-  // the enum values
-  if (schema.enum) {
-    if (schema.enum.every(x => guessType(x) === "number")) {
-      return asNumber(value);
-    } else if (schema.enum.every(x => guessType(x) === "boolean")) {
-      return value === "true";
-    }
   }
-
-  return value;
-}
-
-function getValue(event, multiple) {
   if (multiple) {
     return [].slice
       .call(event.target.options)
       .filter(o => o.selected)
-      .map(o => o.value);
+      .map(o => enumOptions[o.value].value);
   } else {
-    return event.target.value;
+    return enumOptions[event.target.value].value;
+  }
+}
+
+function getOneSelected(value, enumOptions) {
+  if (typeof value === "undefined") {
+    return "";
+  }
+
+  for (const i in enumOptions) {
+    if (_isEqual(enumOptions[i].value, value)) {
+      return i;
+    }
+  }
+  return "";
```

Comment:
Why return an empty string by default?

Could we just return `undefined` instead -- so that it would also be possible to select `""`? (an additional improvement from the old code)

Reply:
sounds like a good idea, will look into it